### PR TITLE
fix(elasticBeanstalkApp): add connection to iamRole, fixes to iamRole

### PR DESCRIPTION
**Pull request by hjaraujof**
## Issue tracker links

[CG-1064](https://autoclouddev.atlassian.net/browse/CG-1064?atlOrigin=eyJpIjoiMjUyMjZjOTE2YTk4NGEwZDlmMDcyZDY1NDc4MWYzMzMiLCJwIjoiaiJ9)

### CODE REVIEW

File: src/services/elasticBeanstalkApplication/schema.graphql [hunk 1]
```diff
@@ -1,8 +1,8 @@
 type awsElasticBeanstalkApp implements awsBaseService @key(fields: "arn") {
   name: String @search(by: [hash, regexp])
   description: String @search(by: [hash, regexp])
-  elasticBeanstalkEnv: [awsElasticBeanstalkEnv] #change to plural
+  iamServiceRole: String @search(by: [hash, regexp])
+  elasticBeanstalkEnvs: [awsElasticBeanstalkEnv]
```
Comment (tyler-dunkel) [inline src/services/elasticBeanstalkApplication/schema.graphql:5]: does this need an update to the `field` for the connection file where this connection is?

Comment (hjaraujof) [reply src/services/elasticBeanstalkApplication/schema.graphql:5]: Yes, it was done. Or do you mean when the connection is formed?

File: src/services/iamRole/connections.ts [hunk 2]
```diff
@@ -229,6 +231,30 @@ export default ({
...
+  const elasticBApps = data.find(
+    ({ name }) => name === services.elasticBeanstalkApp
+  )
+  if (elasticBApps?.data?.[region]) {
+    const dataAtRegion: RawAwsElasticBeanstalkApp[] = elasticBApps.data[
+      region
+    ].filter(
+      ({
+        ResourceLifecycleConfig: { ServiceRole: iamServiceRole } = {},
+      }: RawAwsElasticBeanstalkApp) => iamServiceRole === role.Arn
```
Comment (tyler-dunkel) [inline src/services/iamRole/connections.ts:246]: so If the iam service role is the same, it means the elastic beanstalk app and env are related?

Comment (tyler-dunkel) [reply src/services/iamRole/connections.ts:246]: and wouldnt the id be the id of the elastic beanstalk app not the serviceRole?

Comment (hjaraujof) [reply src/services/iamRole/connections.ts:245]: Answering the first question, the elastic beanstalk app lives in an elastic beanstalk environment. Each time you make changes to an app, it creates a new app version and this app version gets launched in a new environment, meaning that you could potentially switch between environments(or versions).

Regarding the second observation, yes, had a lapsus there, I'm pushing a fix.

**Review by tyler-dunkel — APPROVED** [review]

**Pull request merged**